src/CreatorToken.sol

```diff
@@ -107,10 +152,17 @@ contract CreatorToken is ERC721Royalty {
     if (_referrer != address(0)) _mintAndIncrement(_referrer);
   }
 
+  /// @notice Purchase a token.
+  /// @param _maxPayment The maximum amount of USDC the buyer is willing to pay.
```

Comment:
Should note that the method will revert if the cost to purchase exceeds this amount.

Reply:
added https://github.com/showtime-xyz/creator-tokens/pull/46/commits/3c46721e1c9a0899f54630e2b73028cd5bef159c